docs/design/Ray-Job-Manager.md

```diff
@@ -0,0 +1,78 @@
+# Ray TPU Job Manager
+
+This is a quick design document to explain how our Ray TPU Job Manager works.
+
+## Introduction
+
+Please see the [Ray documentation](https://docs.ray.io/en/latest/index.html) for more information.
+Ray is a resource-aware job scheduler, so you can specify the resources that a job requires:
+
+```python
+@ray.remote(num_cpus=4)
+def my_cpu_job():
+    ...
+```
+
+For GPUs, Ray lets you specify the number of GPUs you need:
+
+```python
+@ray.remote(num_gpus=1)
+def my_gpu_job():
+    ...
+```
+
+In Ray, TPUs are roughly represented the same way, but there are a number of problems with that approach.
+In particular:
+
+* Ray's granularity allows it to schedule a task on a single machine, not across multiple machines. In particular,
+Ray can't directly schedule a task on a TPU slice that spans multiple machines.
+* Google requires that only one process on a machine can access the TPU at a time.
+
+This document explains how we work around those problems.
+
+### Scheduling Slices of TPUs
+
+TPU slices must be used in a SPMD manner (this is probably not quite true, but it's the easiest way to use them).
+This means that you need to schedule the same task on multiple slices of the TPU at once. Ray can't do this directly.
```

Comment:
Terminology: isn't a TPU just one chip?  What does "multiple slices of the TPU mean"?

Reply:
sorry, that was poorly written. I mean you need to schedule the same task on all workers of the same TPU slice at once.